OrbitCore/LinuxTracingHandler.cpp

```diff
@@ -93,11 +95,12 @@ void LinuxTracingHandler::OnCallstack(
   CallStack cs;
   cs.m_ThreadId = callstack.GetTid();
 
+  uint64_t unknownOffset = LinuxTracing::CallstackFrame::kUnknownFunctionOffset;
   for (const auto& frame : callstack.GetFrames()) {
     uint64_t address = frame.GetPc();
     cs.m_Data.push_back(address);
 
-    {
+    if (frame.GetFunctionOffset() == unknownOffset) {
```

Comment:
This is kind-of necessary for frame pointers (at least for now).
The code below is used to cluster samples from the same function, but at different offsets (instructions) together.
Libunwindstack gives us function offset and also the name and the module name directly. In contrast to this, frame pointer based unwinding gives use just the instruction pointers.
We don't know to which function this address belongs to, without reading the Elf file.
But, this would be essentially equivalent to loading all modules in the call chain.

The idea right now would be: For now, we do not cluster instructions from the same function, when using frame pointer based unwinding. 
If the module is loaded, on the "client" side (the UI process), the instruction pointer will be resolved to a name and module information. Otherwise, we will just get the plain address.

My understanding is, that we could do the clustering also in this post processing step, and that we might also think of loading all modules at the beginning of a session. 

Anyway, as framepointer based sampling is disabled for now anyways, it would not hurt us, to keep it that way it is right now. However, I'll add a TODO here.

Reply:
So, if I understand correctly from your description, for DWARF based unwinding (libunwindstack), the function offset would be known here? Why is this not a change of behavior then if you introduce this "if" here?

Reply:
Don't you want to use `!=`?
Also I would refer to `kUnknownFunctionOffset` directly instead of creating the `unknownOffset` for it, which makes it less clear that that's a constant.

Reply:
Got me. Thanks for pointing this out. In order to not change the behavior, it must be `!=` and not `==`. I'll fix it right now.